### PR TITLE
Drop support for Python 3.7

### PR DESCRIPTION
**Pull request by greschd**
Drop support for Python 3.7, and remove it from the tox and CI/CD configurations.

Upgrade mypy to avoid https://github.com/python/mypy/issues/13499, and add checks for 'None' to account for newly reported errors.

### CODE REVIEW

File: src/ansys/dpf/composites/layup_info/_layup_info.py [hunk 1]
```diff
@@ -218,6 +218,9 @@ def get_dpf_material_id_by_analyis_ply_map(
                 layer_index = analysis_ply_info_provider.get_layer_index_by_element_id(
                     element_id_int
                 )
+                assert (
```
Comment (greschd) [inline src/ansys/dpf/composites/layup_info/_layup_info.py:221]: Please check if this assertion makes sense.

File: src/ansys/dpf/composites/layup_info/_layup_info.py [hunk 2]
```diff
@@ -355,6 +358,8 @@ def get_element_info(self, element_id: int) -> Optional[ElementInfo]:
         n_spots = _get_n_spots(apdl_element_type, keyopt_8, keyopt_3)
         dpf_material_ids: Any = []
         element_type = self.dpf_element_types.by_id(element_id)
+        if element_type is None:
```
Comment (greschd) [inline src/ansys/dpf/composites/layup_info/_layup_info.py:361]: Please check if this makes sense.

**Comment by greschd** [comment]
The required 3.7 run will need to be removed in the branch protection rules; instead we can add 3.11.

File: src/ansys/dpf/composites/layup_info/_layup_info.py (hunk 1 above)
Comment (janvonrickenbach) [reply src/ansys/dpf/composites/layup_info/_layup_info.py:221]: Yes that makes sense. I think it should never happen but the typechecker complained.

File: src/ansys/dpf/composites/layup_info/_layup_info.py (hunk 2 above)
Comment (janvonrickenbach) [reply src/ansys/dpf/composites/layup_info/_layup_info.py:361]: Yes makes sense

**Review by janvonrickenbach — APPROVED** [review]

**Pull request merged**